File: jaegr-persistence/src/main/java/com/jaegr/model/GroupView.java
package com.jaegr.model;

import com.jaegr.DBGroup;

import java.util.Set;
import java.util.stream.Collectors;

/**
 * Created by jonas on 30.06.17.
 */
public class GroupView {
    private String name;
    private long id;

    private UserView admin;
    private Set<UserView> users;

    public GroupView(DBGroup group) {
        name = group.getName();
        id = group.getId();
        admin = new UserView(group.getOwner());

        users = group.getUsers().stream()
                .map(u -> new UserView(u))
                .collect(Collectors.toSet());
    }

    public UserView getAdmin() {
        return admin;
    }

    public Set<UserView> getUsers() {
        return users;
    }

    public String getName() {
        return name;
    }

    public long getId() {
        return id;
    }
}

File: jaegr-presentation/src/main/dart/lib/model/GroupView.dart
import 'UserView.dart';
class Group{
  String name;
  int id;
  User admin;
  List<User> members;
  Group (this.name, this.id, this.admin, this.members);
  Group.fromJson(Map json){
    this.name = json["name"];
    this.id = json["id"];
    this.admin = new User.fromJson(json["admin"]);
    this.members = json["users"].map((u) => new User.fromJson(u));

  }
}